handler.go

```diff
@@ -0,0 +1,45 @@
+package fosite
+
+import (
+	"errors"
+	"golang.org/x/net/context"
+	"net/http"
+)
+
+var ErrInvalidResponseType = errors.New("This handler is unable handle any of the response types requested by the auhtorize request")
+var ErrNoResponseTypeHandlerFound = errors.New("None of the handler's are able to handle this authorize request")
+
+type ResponseTypeHandler interface {
+	// HandleResponseType handles an authorize request. To extend the handler's capabilities, the http request
+	// is passed along, if further information retrieval is required.
+	//
+	// If HandleResponseType fails, the handler implementation MUST return ErrInvalidResponseType.
+	HandleResponseType(context.Context, *Response, AuthorizeRequest, http.Request) error
```

Comment:
It's probably a good idea to add `fosite.OAuth2` here as a parameter as well. That way, handlers have access to code generation strategies and much more.